### PR TITLE
Migrate html block test case to Playwright

### PR DESCRIPTION
**Pull request by pooja-muchandikar**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.

Migrate[ html-test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/blocks/html.test.js) to its Playwright version.

## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.

## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.

## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/blocks/html.spec.js`

### CODE REVIEW

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726 

The CI seems to be failing but the error is on the test case which I didn't even alter. `test/e2e/specs/widgets/customizing-widgets.spec.js`

Can you please help me reviewing this PR?

Thanks!

File: test/e2e/specs/editor/blocks/html.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+	test( 'can be created by typing "/html"', async ( { editor, page } ) => {
+		// Create a Custom HTML block with the slash shortcut.
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/html' );
+		await page.locator(
+			`//*[contains(@class, "components-autocomplete__result") and contains(@class, "is-selected") and contains(text(), 'Custom HTML')]`
+		);
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/html.spec.js]: Let's change this to:

```suggestion
		await expect( page.locator( 'role=option[name="Custom HTML"i][selected]' ) ).toBeVisible();
```

**Review by kevin940726 — COMMENTED** [review]
Thanks! The error seems to come from a flaky test. I think we can ignore it :)

**Comment by pooja-muchandikar** [comment]
Hi @kevin940726,

The feedbacks are addressed, please take a look. 

Thanks!

File: test/e2e/specs/editor/blocks/html.spec.js (hunk 1 above)
Comment (pooja-muchandikar) [reply test/e2e/specs/editor/blocks/html.spec.js]: Done.

File: test/e2e/specs/editor/blocks/html.spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+test.describe( 'HTML block', () => {
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/html.spec.js:6]:
```suggestion

test.describe( 'HTML block', () => {
```

File: test/e2e/specs/editor/blocks/html.spec.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/html.spec.js:9]:
```suggestion
	} );

```

File: test/e2e/specs/editor/blocks/html.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		await expect(
+			page.locator( 'role=option[name="Custom HTML"i][selected]' )
+		).toBeVisible();
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '<p>Pythagorean theorem: ' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type(
+			'<var>a</var><sup>2</sup> + <var>b</var><sup>2</sup> = <var>c</var><sup>2</sup> </p>'
+		);
+		// Check the content
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/html.spec.js]:
```suggestion
		// Check the content.
```

**Review by kevin940726 — APPROVED** [review]
Looks great! Thank you again!

Don't mind me, I'm just adding some nitpicking line breaks 😆.

**Comment by kevin940726** [comment]
Oops, I didn't notice that this is a duplicate of https://github.com/WordPress/gutenberg/pull/41127. I think we can still merge this one though, but it'd be nice to check out if the work has already been started by someone next time :)

**Comment by pooja-muchandikar** [comment]
Oh no 😒 sorry I didn't notice that. I had created PR few days back https://github.com/WordPress/gutenberg/pull/40843 but unfortunately I had to close it yesterday and create new one. 

I will take care of this next time 😐

**Pull request merged**